src/nspawn/nspawn.c

```diff
@@ -2162,6 +2163,57 @@ static int setup_boot_id(void) {
         return mount_nofollow_verbose(LOG_ERR, NULL, to, NULL, MS_BIND|MS_REMOUNT|MS_RDONLY|MS_NOSUID|MS_NOEXEC|MS_NODEV, NULL);
 }
 
+static int get_fuse_version(uint32_t *major, uint32_t *minor) {
+        int r;
+        ssize_t n;
+        _cleanup_close_ int fd;
+        _cleanup_(umount_and_rmdir_and_freep) char *dir;
+        _cleanup_free_ char *opts = NULL;
+        union {
+                char unstructured[FUSE_MIN_READ_BUFFER];
+                struct {
+                        struct fuse_in_header header;
+                        /* Don't use <linux/fuse.h>:`struct fuse_init_in` because a newer fuse.h might give
+                         * us a bigger struct than what an older kernel actually gives us, and that would
+                         * break our short-read checks. */
+                        struct {
+                                uint32_t major;
+                                uint32_t minor;
+                        } body;
+                } structured;
+        } request;
+
+        /* Get a FUSE handle. */
+        fd = open("/dev/fuse", O_RDWR);
```

Comment:
as per coding style, we require that all open() calls use O_CLOEXEC

Reply:
Fixed.